### ISSUE TITLE
Wrong Subsumption

### ISSUE THREAD

**Issue by domainexpert** [tracer-x/klee#146]
Discovered for `basic/addition_safe8.c` in tracer-x/klee-examples@bcf62f9. The following shows the tree in some previous version:
![addition_safe8](https://user-images.githubusercontent.com/696558/29258378-a85637ec-80eb-11e7-813d-c770bc4b3672.png)
Whereas this is the new tree:
![addition_safe8_new](https://user-images.githubusercontent.com/696558/29258383-b4f2a418-80eb-11e7-81b0-0a89ee41518d.png)

**EDIT: Added actual pictures**

**Comment by domainexpert** [comment]
The new tree above is probably the correct one. This is because there should not be any subsumption at the intermediate nodes. This is the tree of the latest revision (8 November 2017):

![tree](https://user-images.githubusercontent.com/696558/32531271-d1a254ca-c47d-11e7-81f3-cf61650f8ee9.png)

There are three subsumptions: the intermediate one of 13 by 6 is suspicious. Following is supposedly their constraints:
```
6: (exists x0 . x = x0 + 3, x0 <= 0) <=> (x <= 3)
13: (exits x0 . x = x0 + 2 + y0, x0 <= 0, 0 <= y0 <= 35) <=> (x <= 37)
```

**Comment by domainexpert** [comment]
Resolved via #324.
